src/west/configuration.py

```diff
@@ -647,59 +688,66 @@ def _location(cfg: ConfigFile, topdir: PathType | None = None, find_local: bool
-            return os.fspath(pd / 'west' / 'config')
+            return [os.fspath(pd / 'west' / 'config')]
 
         raise ValueError('unsupported platform ' + plat)
     elif cfg == ConfigFile.GLOBAL:
         if 'WEST_CONFIG_GLOBAL' in env:
-            return _rel_topdir_to_abs(env['WEST_CONFIG_GLOBAL'], topdir)
+            paths = str_to_paths(env['WEST_CONFIG_GLOBAL'])
+            return [_rel_topdir_to_abs(p, topdir) for p in paths]
 
         if platform.system() == 'Linux' and 'XDG_CONFIG_HOME' in env:
-            return os.path.join(env['XDG_CONFIG_HOME'], 'west', 'config')
+            return [os.path.join(env['XDG_CONFIG_HOME'], 'west', 'config')]
 
-        return os.fspath(Path.home() / '.westconfig')
+        return [os.fspath(Path.home() / '.westconfig')]
     elif cfg == ConfigFile.LOCAL:
         if 'WEST_CONFIG_LOCAL' in env:
-            return _rel_topdir_to_abs(env['WEST_CONFIG_LOCAL'], topdir)
+            paths = str_to_paths(env['WEST_CONFIG_LOCAL'])
+            return [_rel_topdir_to_abs(p, topdir) for p in paths]
 
         if topdir:
-            return os.fspath(Path(topdir) / WEST_DIR / 'config')
+            return [os.fspath(Path(topdir) / WEST_DIR / 'config')]
 
         if find_local:
             # Might raise WestNotFound!
-            return os.fspath(Path(west_dir()) / 'config')
+            return [os.fspath(Path(west_dir()) / 'config')]
         else:
-            return ''
+            return []
     else:
         raise ValueError(f'invalid configuration file {cfg}')
 
 
-def _gather_configs(cfg: ConfigFile, topdir: PathType | None) -> list[str]:
+def _gather_configs(cfg: ConfigFile, topdir: PathType | None) -> list[Path]:
     # Find the paths to the given configuration files, in increasing
     # precedence order.
-    ret = []
+    ret: list[Path] = []
 
     if cfg == ConfigFile.ALL or cfg == ConfigFile.SYSTEM:
-        ret.append(_location(ConfigFile.SYSTEM, topdir=topdir))
+        paths = _location(ConfigFile.SYSTEM, topdir=topdir)
+        ret.extend(str_list_to_paths(paths))
     if cfg == ConfigFile.ALL or cfg == ConfigFile.GLOBAL:
-        ret.append(_location(ConfigFile.GLOBAL, topdir=topdir))
+        paths = _location(ConfigFile.GLOBAL, topdir=topdir)
+        ret.extend(str_list_to_paths(paths))
     if cfg == ConfigFile.ALL or cfg == ConfigFile.LOCAL:
         try:
-            ret.append(_location(ConfigFile.LOCAL, topdir=topdir))
+            paths = _location(ConfigFile.LOCAL, topdir=topdir)
+            ret.extend(str_list_to_paths(paths))
         except WestNotFound:
             pass
-
     return ret
 
 
 def _ensure_config(configfile: ConfigFile, topdir: PathType | None) -> str:
     # Ensure the given configfile exists, returning its path. May
     # raise permissions errors, WestNotFound, etc.
-    loc = _location(configfile, topdir=topdir)
-    path = Path(loc)
-
+    config_file = str_list_to_paths(_location(configfile, topdir=topdir))
```

Comment:
We can't have both `config_file` and `configfile` in the same function... `configs` maybe?